src/components/RangeSlider/components/SingleThumb/SingleThumb.scss

```diff
@@ -91,19 +142,24 @@ $range-thumb-shadow-focus: 0 0 0 rem(1px) color('indigo');
     cursor: -webkit-grab;
     width: $range-thumb-size;
     height: $range-thumb-size;
-    border: border-width() solid color('sky', 'lighter');
+    border: border-width() solid var(--p-interactive, color('sky', 'lighter'));
     border-radius: 50%;
-    background: linear-gradient(color('white'), color('sky', 'lighter'));
+    background: var(
+      --p-interactive,
+      linear-gradient(color('white'), color('sky', 'lighter'))
+    );
     box-shadow: $range-thumb-shadow;
     appearance: none;
-    transition-property: border-color, box-shadow;
+    transition-property: width, height, margin-top, border-color, box-shadow;
     transition-duration: duration();
     transition-timing-function: easing();
 
+    @include rangeSliderThumbPosition($range-thumb-size);
+
     &:hover {
-      background: linear-gradient(
-        color('sky', 'lighter'),
-        color('sky', 'light')
+      background: var(
+        --p-interactive,
+        linear-gradient(color('sky', 'lighter'), color('sky', 'light'))
```

Comment:
Fallback values using `linear-gradient` trigger the Safari infinite re-render bug. This will instead need to be done by conditionally adding a class with the new value, or another way which avoids the gradient fallback.

The `useFeatures` hook can be used to know if global theming is enabled or not.

Reply:
I wonder if we can use the fallback values within the `linear-gradient` like,

```scss
background: linear-gradient(
  var(--p-interactive, color('sky', 'lighter')),
  var(--p-interactive, color('sky', 'light'))
);
```

Reply:
I can’t remember, fairly easy to test in Safari though. Happy to walk you through it